src/gql/GQLWrapper.tsx

```diff
@@ -1,64 +1,48 @@
 import React, { useState, useEffect } from "react";
-import { ApolloClient } from "apollo-client";
-import { ApolloProvider } from "@apollo/react-hooks";
-import { HttpLink } from "apollo-link-http";
-import { ApolloLink } from "apollo-link";
-import { RetryLink } from "apollo-link-retry";
-import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
 import {
-  addMockFunctionsToSchema,
-  introspectSchema,
-  makeExecutableSchema,
-} from "graphql-tools";
-import { printSchema } from "graphql/utilities/schemaPrinter";
-import { SchemaLink } from "apollo-link-schema";
-import { onError } from "apollo-link-error";
+  ApolloClient,
+  ApolloProvider,
+  InMemoryCache,
+  ApolloLink,
+  HttpLink,
+} from "@apollo/client";
+import { RetryLink } from "@apollo/client/link/retry";
+import { onError } from "@apollo/client/link/error";
 import { useAuthDispatchContext, Logout, Dispatch } from "context/auth";
-import ApolloLinkTimeout from "apollo-link-timeout";
 import { reportError } from "utils/errorReporting";
+import { getGQLUrl } from "utils/getEnvironmentVariables";
+
+const GQLWrapper: React.FC = ({ children }) => {
+  const [client, setClient] = useState(null);
+  const { logout, dispatch } = useAuthDispatchContext();
+
+  useEffect(() => {
+    async function getAndSetClient(): Promise<void> {
+      const gqlClient = await getGQLClient({
```

Comment:
`getGQLClient` is not longer async so we can don't need to import `useState` in this file

Reply:
yeah i thought so too which is why my original refactor was `const client = new ApolloClient()`. doing const client = new ApolloClient()  breaks authentication, which breaks e2e tests. the error is caused by infinite logout requests to the server, which are caused by infinite initial requests from the initial load logic. 

asynchronously setting client in state prevents the infinite rerenders.

Reply:
ah, I see what you mean by the infinite renders when we dont save the result of getGQLClient in state. We can actually do something like this since `getGQLClient` function is not asynchronous and shouldn't follow the `await` operator. 

```  const [client] = useState(
    getGQLClient({
      credentials: "include",
      gqlURL: getGQLUrl(),
      logout,
      dispatch,
    })
  );```

Reply:
there were consistent flakes for several retries when i changed it to your suggestion. when i reverted it back to my original implementation, the flakes stopped. the change you propose is cleaner but has no technical difference and, for whatever reason, created more flakes. I am fine merging it as my original implementation.

Reply:
can we remove the await on line 21 since `getGQLClient` is not an async function? I created a ticket to cleanup the rest of GQLWrapper since we've accumulated some tech debt here. https://jira.mongodb.org/browse/EVG-12889